### DISCUSSION TITLE
How to use/edit repositories in Github desktop

### DISCUSSION THREAD

**Discussion by misterskye** [earthlab-education/Earth-Analytics-2023-01-Intro#318] [! GitHub help]
Next week I will be in the field, and I'm trying to prepare for periods of no internet.  Consequently, I'm wondering if you have a recipe for offline work.  I downloaded Github desktop, but would like some best practices.

Thanks

**Comment by eculler** [comment]
GitHub desktop is a great option! But keep in mind that you won't be able to actually interface with GitHub without internet.

Best practices for working without internet: 
  * Install the [earth-analytics-python environment on your computer](https://earthlab-education.github.io/Earth-Analytics-2023-01-Intro/python-open-source-tool-setup/setup-02-setup-git-bash-and-anaconda.htm)
  * You can also [install VSCode for your desktop](https://code.visualstudio.com/download) if you prefer that to Jupyter Notebook or Jupyter Lab
  * Try to get your data downloaded before you leave, or you'll be stuck. You could even [download some data manually from AppEEARS](https://appeears.earthdatacloud.nasa.gov/task/area) if you want to work on the analysis part and haven't finished the download.

As far as best practices, they aren't so different from working on Codespaces:
  1. Always pull and merge before working, if you have internet access
  2. Commit early and often, leaving detailed messages for yourself in case you need to revert. For sure when you have finished a work session, committing and pushing is a good idea (again dependent on internet access). But also if you plan to make some major changes, committing is a good idea to bookmark a working version
  3. If you're collaborating, make sure to work in a branch if you're a regular contributor, or a fork if you're not.

The key one is making sure to pull and merge changes before you start working, which won't be possible without internet. But in general you will want to think about it! For your assignment repositories, you're the only one working on them so it shouldn't be a problem. But for the Earth-Analytics-2023-01-Intro repository it's important, just like it's important to pull in changes from main before you make changes to try to avoid conflicts.
